resource.h

```diff
@@ -0,0 +1,163 @@
+/* Copyright (C) 2018, Oracle and/or its affiliates. All rights reserved
+ *
+ * This file is part of libresource.
+ *
+ * libresource is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * libresource is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with libresource. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * This is main header file which declares all resource ids and interfaces.
+ *
+ */
+
+#ifndef	_RESOURCE_H
+#define	_RESOURCE_H
+
+#include <net/if.h>
+
+/* libresource version */
+#define LIBRESOURCE_API_VERSION (1.0.0)
+
+/* Possible status for libresource information returned */
+/* libresource information was fetched correctly */
+#define RES_STATUS_FILLED 0
+/* There was some error in fetching libresource information. In most cases
+ * errno will be set.
+ */
+#define RES_STATUS_EMPTY -1
+/* Resource information is not supported yet, or Invalid resource
+ * information.
+ */
+#define RES_STATUS_NOTSUPPORTED	-2
+/* If partial information was read for a libresource information. For example
+ * a string was read partially.
+ */
+#define RES_STATUS_TRUNCATED -3
+
+
+/* Maximum size of a resource information data type which can be returned
+ * without explicitly allocating memory for it. If resource information
+ * size is larger than this this a pointer to allocated memory will be
+ * returned.
+ */
+#define RES_UNIT_OUT_SIZE	256
+
+/* This union is used to return resource information of various types */
+union r_data {
+	int i;
+	size_t sz;
+	long l;
+	char str[RES_UNIT_OUT_SIZE];
```

Comment:
What exactly is this for?

Reply:
The union itself is pretty opaque right now. Can you expand what the individual fields are for, please?

Reply:
do you mean to put comments and explain what resource IDs will use what fields ?
This union is to provide user a way to get info without having to typecase return info in userland and was one of the suggestions on my draft. one field will can be used for multiple resoure IDs, for example both no of CPUs and no of Interfaces can be returned as integar.